pycloudlib/ibm_classic/cloud.py

```diff
@@ -0,0 +1,500 @@
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""IBM Cloud type."""
+
+from typing import List, Literal, Optional, Tuple
+
+import SoftLayer  # type: ignore
+
+from pycloudlib.cloud import BaseCloud
+from pycloudlib.config import ConfigFile
+from pycloudlib.ibm_classic.errors import IBMClassicException
+from pycloudlib.ibm_classic.instance import IBMClassicInstance
+from pycloudlib.instance import BaseInstance
+
+
+class IBMClassic(BaseCloud):
+    """IBM Classic Class."""
+
+    _type = "ibmclassic"
+
+    def __init__(
+        self,
+        tag: str,
+        timestamp_suffix: bool = True,
+        config_file: Optional[ConfigFile] = None,
+        *,
+        sl_username: Optional[str] = None,
+        sl_api_key: Optional[str] = None,
+    ):
+        """Initialize base cloud class.
+
+        Args:
+            tag: string used to name and tag resources with
+            timestamp_suffix: Append a timestamped suffix to the tag string.
+            config_file: path to pycloudlib configuration file
+            sl_username: IBM Softlayer username
+            sl_api_key: IBM Softlayer API key
+        """
+        super().__init__(
+            tag,
+            timestamp_suffix,
+            config_file,
+            required_values=[sl_username, sl_api_key],
+        )
+        self.created_keys: List[str] = []
+        self.created_security_groups: list = []
+
+        self._sl_username = sl_username or self.config.get("sl_username")
+        self._sl_api_key = sl_api_key or self.config.get("sl_api_key")
+
+        self._log.debug("logging into IBM")
+
+        self._client = SoftLayer.create_client_from_env(
+            username=self._sl_username,
+            api_key=self._sl_api_key,
+        )
+        self._virtual_server_manager = SoftLayer.VSManager(client=self._client)
+        self._image_manager = SoftLayer.ImageManager(client=self._client)
+        self._ssh_key_manager = SoftLayer.SshKeyManager(client=self._client)
+        self._network_manager = SoftLayer.NetworkManager(client=self._client)
+
+    def delete_image(self, image_id: str, **kwargs):
+        """Delete an image.
+
+        Args:
+            image_id: string, id of the image to delete
+            **kwargs: dictionary of other arguments to pass to delete_image
+        """
+        self._image_manager.delete_image(int(image_id))
```

Comment:
thank for catching this!

Reply:
okay so i changed image_id to a int in the funtion signature and mypy is now throwing a fit :(

Reply:
I do not see any mypy error with:

```suggestion
        self._image_manager.delete_image(image_id)
```

Could we change it to this? Otherwise, this going to error if image_id is not convertible to int.